File: src/android/firebase_analytics/src/main/java/com/ee/FirebaseEventParameter.kt
package com.ee

internal object FirebaseEventType {
    const val Long = 0
    const val Double = 1
    const val String = 2
}

internal class FirebaseEventParameter(
    val _type: Int,
    val _value: String) {
    val type = _type
    val longValue = _value.toLong()
    val doubleValue = _value.toDouble()
    val stringValue = _value
}

File: src/ios/ee/firebase_analytics/FirebaseEventParameter.swift
//
//  FirebaseEventParameter.swift
//  ee-x-a16b5188
//
//  Created by eps on 2/1/21.
//

enum FirebaseEventType: Int {
    case Long
    case Double
    case String
}

internal class FirebaseEventParameter {
    private let _type: FirebaseEventType
    private let _value: String

    public init(_ type: Int, _ value: String) {
        _type = FirebaseEventType(rawValue: type) ?? FirebaseEventType.Long
        _value = value
    }

    public var longValue: Int64 {
        return Int64(_value) ?? 0
    }

    public var doubleValue: Double {
        return Double(_value) ?? 0
    }

    public var stringValue: String {
        return _value
    }
}
